src/api/repos/activity-repo.ts

```diff
@@ -24,4 +26,36 @@ const getActivity = async (id: number) => {
     };
 };
 
-export { getAllActivities, getActivity };
+/** Search for Activities using the provided 'keyword' */
+const searchActivities = async (keyword: string) => {
+    // Repack the keyword string to be usable in pgsql to_tsquery()
+    // And we don't want to use phraseto_tsquery() because we don't need to match the order of words
+    const keywordListRepacked = keyword
+        .split(' ')
+        .filter((item) => {
+            return item.trim().length > 1; // Remove meaningless empty or 1-letter words
+        })
+        .join(' & ');
+
+    const MAX_SEARCH_RESULT = 30;
+    const queryString =
+        // QUERY TO SEARCH FOR ACTIVITIES USING FULL-TEXT SEARCH
+        `SELECT A."id", "title", "description", "images"[1]
+         FROM "Activities_Tokens_Search" ATS,
+              "Activities" A
+         WHERE 
+           ATS."tokens" @@ to_tsquery('english', '${keywordListRepacked}')
```

Comment:
SQL injection alert 🚨🚨🚔 
I'm not a pro hacker but I'm pretty sure that someone could use this to do bad thing. 

<img width="1014" alt="Screenshot 2023-02-18 at 9 41 09 AM" src="https://user-images.githubusercontent.com/29782830/219871848-2840bb10-d747-4642-b273-bf21913dabb0.png">

Reply:
The keyword is converted into vector with `to_tsquery` so I don't think it will create any (clearly) obvious problem. There could be underlying problems, I admit, and yes I should filter the input string better. Will be on it.

Reply:
Turns out the problem is not a problem. The worst they can do is erase the other query that they themselves entered. It just defeats the purpose of entering other parameters but does not affect the application like I thought earlier. If they go hardcore sql injection they will received an error.
<img width="1009" alt="Screenshot 2023-02-19 at 5 30 22 PM" src="https://user-images.githubusercontent.com/29782830/219979388-047d9646-50a5-4895-a7c8-f165ed5e32a9.png">
<img width="1008" alt="Screenshot 2023-02-19 at 5 30 35 PM" src="https://user-images.githubusercontent.com/29782830/219979391-d7e80c03-2e14-4db6-bb00-ce58eabdb628.png">

Reply:
I guess it's more of a user problem